src/or/shared_random_common.c

```diff
@@ -0,0 +1,258 @@
+/* Copyright (c) 2018, The Tor Project, Inc. */
+/* See LICENSE for licensing information */
+
+/**
+ * \file shared_random_common.c
```

Comment:
I don't think "common" is the right name for this kind of module.  This isn't exactly "the parts of the API that are shared by some other things" -- it's "the parts of the API that are used even if you aren't a dirauth."  Maybe "shared_random_user" or "shared_random_client"?  Hm.  Or maybe something else.  This is tricky.

Reply:
Yes tricky, we had a bikeshed moment but I'm still unsure to this moment...

Bottom line is that they are functions that other non-dirauth part of Tor needs. I like the `_client.c` idea. For us developers, it makes sense but looking from another angle, all the functions in those two `_common.h` files are only for HS.

Actually, everything in `dirvote_common.h` is for `shared_random_common.c`  and then that file is for hidden service.

There could be an argument to actually isolate all of this in one hidden service file like `hs_shared_random.c` and `dirvote_common.c` into that file or in a new specific file?

Reply:
I think we can rename these post-merge, but we should figure out what we're renaming these to.

Reply:
Agree. The more we talk about it and the more I don't like `_common.c`.